xbmc/network/Network.cpp

```diff
@@ -142,10 +142,7 @@ CNetwork::CNetwork()
   CApplicationMessenger::GetInstance().PostMsg(TMSG_NETWORKMESSAGE, SERVICES_UP, 0);
 }
 
-CNetwork::~CNetwork()
-{
-  CApplicationMessenger::GetInstance().PostMsg(TMSG_NETWORKMESSAGE, SERVICES_DOWN, 0);
```

Comment:
I am not convinced that the event can be dropped without side effects.

Reply:
CApplication sends the same message inside StartServices method
sent message at app cleanup will not delivered in any case.

Reply:
see https://github.com/xbmc/xbmc/blob/master/xbmc/Application.cpp#L1276

Reply:
That does not matter. CNetwork has to be treated as a self-contained service that can be started and stopped separately if needed.
plaese elaborate why this should fix anything.

Reply:
why `CNetwork` has to be treated as a self-contained service if its instance is hold by `CServiceManager` and it's a singleton.
I didn't find a possibility to re-initialize `CNetwork` service and why it may occurs.

Reply:
fixed in another way.

Reply:
thanks.

> why CNetwork has to be treated as a self-contained service if its instance is hold by CServiceManager and it's a singleton.

SeriveManager is an intermediate solution on road away from globals.